ext/ffi/00_ffi.js

```diff
@@ -326,6 +334,33 @@
             },
           );
         }
+
+        if (needsUnpacking && !isNonBlocking) {
+          const call = this.symbols[symbol];
+          const parameters = symbols[symbol].parameters;
+          const vi = new Int32Array(2);
+          const vui = new Uint32Array(vi.buffer);
+          const b = new BigInt64Array(vi.buffer);
+
+          const params = parameters.map((_, index) => `p${index}`);
+          // Make sure V8 has no excuse to not optimize this function.
+          this.symbols[symbol] = new Function(
+            "vi",
+            "vui",
+            "b",
+            "call",
+            `return function (${params.join(", ")}) {
+            call(${params.join(", ")}${parameters.length > 0 ? ", " : ""}vi);
+            ${
+              isI64(resultType)
+                ? `const n1 = Number(b[0])`
+                : `const n1 = vui[0] + 2 ** 32 * vui[1]` // Faster path for u64
+            };
+            if (Number.isSafeInteger(n1)) return n1;
```

Comment:
Ouch, can we use primordials here for `Number`, `NumberIsSafeInteger`, and such?